src/parser.y

```diff
@@ -357,9 +357,8 @@ Term "as" Patterns '|' Exp {
 "if" Exp "then" Exp ElseBody {
   $$ = gen_cond($2, $4, $5);
 } |
-"if" Exp "then" error {
```

Comment:
I think we still want to retain an error case though, for when the `end` token is missing.

Reply:
Good point.

Reply:
Fixed in latest commit. (Can squash when everything looks good)